gobblin-cluster/src/main/java/gobblin/cluster/GobblinClusterManager.java

```diff
@@ -113,7 +112,8 @@
 
   protected final String applicationId;
 
-  public GobblinClusterManager(String clusterName, String applicationId, Config config) throws Exception {
+  public GobblinClusterManager(String clusterName, String applicationId, Config config,
+      Optional<Path> appWorkDirOptional) throws Exception {
```

Comment:
The work dir was being specified in gobblin-yarn config but not being used anywhere. This helps us make use of it if specified, though gobblin-yarn still needs to be fixed for it. There was also an issue around it reported by someone on google groups for gobblin-yarn